cnd/src/swap_protocols/facade.rs

```diff
@@ -51,43 +51,62 @@ use std::{convert::TryInto, fmt::Debug, sync::Arc};
 pub struct Facade {
     pub bitcoin_connector: Arc<btsieve::bitcoin::Cache<BitcoindConnector>>,
     pub ethereum_connector: Arc<ethereum::Cache<Web3Connector>>,
-    pub state_store: Arc<InMemoryStateStore>,
+    pub alpha_ledger_state: Arc<AlphaLedgerState>,
+    pub beta_ledger_state: Arc<BetaLedgerState>,
+    pub swap_communication_state: Arc<SwapCommunicationState>,
+    pub swap_error_state: Arc<SwapErrorState>,
     pub seed: RootSeed,
     pub swarm: Swarm,
     pub db: Sqlite,
 }
 
-impl<S> Insert<S> for Facade
+impl<AL, BL, AA, BA, AI, BI> state::Insert<SwapCommunication<AL, BL, AA, BA, AI, BI>> for Facade
 where
-    S: Send + 'static,
+    AL: Clone + Send + 'static,
```

Comment:
:mega: I think these should not be necessary (esp. `Clone`) and if, then it should be enough to put the bounds on `SwapCommunication<AL, BL, AA, BA, AI, BI>`.

Reply:
Good catch, `Clone` is not needed for `Insert`.

Reply:
`Send` and `'static` are needed, but can be applied to the combined type.